presto-main-base/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -947,8 +948,10 @@ public PlanOptimizers(
         // MergeJoinForSortedInputOptimizer can avoid the local exchange for a join operation
         // Should be placed after AddExchanges, but before AddLocalExchange
         // To replace the JoinNode to MergeJoin ahead of AddLocalExchange to avoid adding extra local exchange
+        // SortedExchangeRule pushes sorts down to exchange nodes for distributed queries
         builder.add(new MergeJoinForSortedInputOptimizer(metadata, featuresConfig.isNativeExecutionEnabled()),
-                new SortMergeJoinOptimizer(metadata, featuresConfig.isNativeExecutionEnabled()));
+                new SortMergeJoinOptimizer(metadata, featuresConfig.isNativeExecutionEnabled()),
+                new SortedExchangeRule(metadata));
```

Comment:
Shall we gate this optimizer only for Presto-on-Spark? we can leverage `featuresConfig.isPrestoSparkExecutionEnvironment()`.